### PR TITLE
feature(*) Migrate to Cicero 0.22 & Concerto 1.0

### PR DESCRIPTION
**Pull request by jeromesimeon**
### Changes

- Updates templates for Cicero `0.22` with Ergo `0.22` and Concerto `1.0`

### Flags

- ~Only around half the templates have been migrated, this PR is open as Draft for documentation purposes~ All templates now migrated

### CODE REVIEW

File: src/docusign-connect/logic/logic.ergo [hunk 1]
```diff
@@ -33,7 +34,6 @@ contract DocuSignEnvelopeCounter over DocuSignEnvelopeCounterClause state DocuSi
   // initialize the state counter
   clause init() : Response {
     set state DocuSignEnvelopeCounterState {
-      stateId: "com.docusign.connect.clause.DocuSignEnvelopeCounterState",
```
Comment (mttrbrts) [inline src/docusign-connect/logic/logic.ergo:36]: 🥳

File: src/docusign-po-failure/logic/logic.ergo [hunk 2]
```diff
@@ -120,7 +119,7 @@ contract PurchaseOrderFailureContract over PurchaseOrderFailure state PurchaseOr
       promisee: some(contract.buyer),
       deadline: none,
       amount: penalty,
-      description: contract.buyer.partyId ++ " should be paid a penalty"
+      description: toString(contract.buyer) ++ " should be paid a penalty"
```
Comment (mttrbrts) [inline src/docusign-po-failure/logic/logic.ergo:122]: What is the value of `toString(reference)`? Something like `resource:namespace.Type#identifier`?

File: src/ip-payment/test/logic.feature [hunk 3]
```diff
@@ -6,6 +6,7 @@ Feature: IP Payment Contract
 
   Scenario: Payment of a specified amount should be made
     When the current time is "2019-03-04T16:34:00-05:00"
+    And the UTC offset is -5
```
Comment (mttrbrts) [inline src/ip-payment/test/logic.feature:9]: This is neat!

File: src/perishable-goods/test/logic.feature [hunk 4]
```diff
@@ -75,16 +71,15 @@ Formula for Breach Penalty Calculation:
...
+    "$class": "org.accordproject.obligation.PaymentObligation",
     "amount": {
       "$class": "org.accordproject.money.MonetaryAmount",
       "doubleValue": 300,
       "currencyCode": "USD"
     },
-    "description": "DAN should pay shipment amount to PETER",
-    "promisor": "resource:org.accordproject.cicero.contract.AccordParty#DAN",
-    "promisee": "resource:org.accordproject.cicero.contract.AccordParty#PETER",
-    "eventId": "valid"
```
Comment (mttrbrts) [inline src/perishable-goods/test/logic.feature:87]: `eventId` is replaced by `$identifier` now, right?

File: src/servicelevelagreement/logic/logic.ergo [hunk 5]
```diff
@@ -39,23 +38,24 @@ define function createPaymentObligation(
       amount: amount,
       description:
         "payment owed by "
-          ++ serviceProvider.partyId
-          ++ " to " ++ serviceConsumer.partyId
+					++ toString(serviceProvider)
```
Comment (mttrbrts) [inline src/servicelevelagreement/logic/logic.ergo:41]: Some weird indentation?

**Review by mttrbrts — APPROVED** [review]
Wow, super speedy! I was planning to help with this, but you're just too fast!

I've added a few minor observations, but nothing blocking.

The diff tally is interesting ...

> "Cicero 0.22: now with 33% less code!"

File: src/perishable-goods/test/logic.feature (hunk 4 above)
Comment (jeromesimeon) [reply src/perishable-goods/test/logic.feature:87]: Correct: added by the validator so shows in the output:
```
zsh-5.8$ cd ../payment-upon-delivery/
zsh-5.8$ cicero trigger
6:51:11 AM - INFO: Using current directory as template folder
6:51:11 AM - INFO: Loading a default text/sample.md file.
6:51:11 AM - INFO: Loading a default request.json file.
6:51:13 AM - WARN: A state file was not provided, initializing state. Try the --state flag or create a state.json in the root folder of your template.
6:51:13 AM - INFO: 
{
  "clause": "payment-upon-delivery@0.12.0-ff3f68ca77f73bfa73969884d5fc6be1592154e9aae2ef11a67cb66f87fb0f59",
  "request": {
    "$class": "org.accordproject.payment.upondelivery.DeliveryAcceptedRequest"
  },
  "response": {
    "$class": "org.accordproject.payment.upondelivery.DeliveryAcceptedResponse",
    "$timestamp": "2021-04-14T06:51:13.265-04:00"
  },
  "state": {
    "$class": "org.accordproject.runtime.State",
    "$identifier": "010187ba-ef2b-490f-8836-92dc4be6e1a6"
  },
  "emit": [
    {
      "$class": "org.accordproject.obligation.PaymentObligation",
      "amount": {
        "$class": "org.accordproject.money.MonetaryAmount",
        "doubleValue": 11.98,
        "currencyCode": "USD"
      },
      "description": ""resource:org.accordproject.contract.Party#Michael" should pay cost of goods and delivery fee to "resource:org.accordproject.contract.Party#Diana"",
      "$identifier": "5c3bd7c5-bd2b-48b8-ad5b-ed0d46d24d7b",
      "contract": "resource:org.accordproject.payment.upondelivery.PaymentUponDeliveryContract#2ca304c5-f9f0-4d93-86a9-5095d0a4726c",
      "promisor": "resource:org.accordproject.contract.Party#Michael",
      "promisee": "resource:org.accordproject.contract.Party#Diana",
      "$timestamp": "2021-04-14T06:51:13.266-04:00"
    }
  ]
}
```

File: src/ip-payment/test/logic.feature (hunk 3 above)
Comment (jeromesimeon) [reply src/ip-payment/test/logic.feature:9]: I think it works. Reminds me I need to add that to the CLIs too.

File: src/docusign-po-failure/logic/logic.ergo (hunk 2 above)
Comment (jeromesimeon) [reply src/docusign-po-failure/logic/logic.ergo:122]: yeah. Relationships are still a pretty murky area from and Ergo standpoint, needs re-review.

**Review by dselman — APPROVED** [review]
LKTM! Nice work.

File: src/servicelevelagreement/logic/logic.ergo (hunk 5 above)
Comment (jeromesimeon) [reply src/servicelevelagreement/logic/logic.ergo:41]: Good catch. I really need to figure out tabbing configuration for syntax editing.

**Pull request merged**